src/main/core/scheduler/scheduler_policy_host_single.c

```diff
@@ -286,6 +286,35 @@ static Event* _schedulerpolicyhostsingle_pop(SchedulerPolicy* policy, Simulation
     return NULL;
 }
 
+static Event* _schedulerpolicyhostsingle_peekHost(SchedulerPolicy* policy, Host* host) {
+    MAGIC_ASSERT(policy);
+    HostSinglePolicyData* data = policy->data;
+
+    /* figure out which hosts we should be checking */
+    HostSingleThreadData* tdata =
+        g_hash_table_lookup(data->threadToThreadDataMap, GUINT_TO_POINTER(pthread_self()));
+    utility_assert(tdata);
+
+    HostSingleQueueData* qdata = g_hash_table_lookup(data->hostToQueueDataMap, host);
+    utility_assert(qdata);
+
+    /* FIXME split  this timer? */
+#ifdef USE_PERF_TIMERS
+    /* tracking idle time spent waiting for the host queue lock */
+    g_timer_continue(tdata->popIdleTime);
+#endif
+    g_mutex_lock(&(qdata->lock));
+#ifdef USE_PERF_TIMERS
+    g_timer_stop(tdata->popIdleTime);
+#endif
+
+    Event* nextEvent = priorityqueue_peek(qdata->pq);
+
+    g_mutex_unlock(&(qdata->lock));
+
+    return nextEvent;
```

Comment:
Hmm, my thinking was that `qdata->lock` here is only needed to protect the hash table itself, not the individual host queue or events in it; those are protected by the Host lock.

That is perhaps a bit subtle, though. Since we really only need the time of the next event I'll change this interface to just return that.

Reply:
Changed to return the time instead of the Event.